Sources/Subscription/Flows/AppStore/AppStorePurchaseFlow.swift

```diff
@@ -108,14 +110,41 @@ public final class AppStorePurchaseFlow {
             }
         }
     }
-    // swiftlint:enable cyclomatic_complexity
 
+    // swiftlint:enable cyclomatic_complexity
     @discardableResult
-    public static func completeSubscriptionPurchase() async -> Result<PurchaseUpdate, AppStorePurchaseFlow.Error> {
+    public static func completeSubscriptionPurchase(with transactionJWS: TransactionJWS) async -> Result<PurchaseUpdate, AppStorePurchaseFlow.Error> {
         os_log(.info, log: .subscription, "[AppStorePurchaseFlow] completeSubscriptionPurchase")
 
-        let result = await AccountManager.checkForEntitlements(wait: 2.0, retry: 20)
+        guard let accessToken = AccountManager().accessToken else { return .failure(.missingEntitlements) }
+
+        let result = await callWithRetries(wait: 0.5, retry: 3) {
+            switch await SubscriptionService.confirmPurchase(accessToken: accessToken, signature: transactionJWS) {
+            case .success:
+                return true
+            case .failure:
+                return false
+            }
+        }
 
         return result ? .success(PurchaseUpdate(type: "completed")) : .failure(.missingEntitlements)
     }
+
+    private static func callWithRetries(wait waitTime: Double, retry retryCount: Int, conditionToCheck: () async -> Bool) async -> Bool {
```

Comment:
WDYT about moving the retry logic to `executeAPICall`in the APIService?   That way, we could easily implement retry in every Backend call.

Reply:
I think it is worth considering, though I would defer it until we have a need for other calls to be made with retires.